sycl/include/sycl/ext/intel/experimental/esimd/memory.hpp

```diff
@@ -2854,8 +2854,13 @@ template <__ESIMD_NS::atomic_op Op, typename T, int N,
           cache_hint L1H = cache_hint::none, cache_hint L3H = cache_hint::none,
           typename AccessorTy, typename Toffset>
 __ESIMD_API std::enable_if_t<
-    sycl::detail::acc_properties::is_accessor_v<AccessorTy> &&
-        !sycl::detail::acc_properties::is_local_accessor_v<AccessorTy>,
+    (__ESIMD_DNS::is_device_accessor_with_v<
+         AccessorTy, __ESIMD_DNS::accessor_mode_cap::can_read> &&
+     Op == __ESIMD_NS::atomic_op::load) ||
+        (__ESIMD_DNS::is_device_accessor_with_v<
+             AccessorTy, __ESIMD_DNS::accessor_mode_cap::can_read> &&
+         __ESIMD_DNS::is_device_accessor_with_v<
+             AccessorTy, __ESIMD_DNS::accessor_mode_cap::can_write>),
```

Comment:
nit: this could be  a bit shorter - read permission is required for all operations.
```suggestion
    (__ESIMD_DNS::is_device_accessor_with_v<
         AccessorTy, __ESIMD_DNS::accessor_mode_cap::can_read> &&
     (Op == __ESIMD_NS::atomic_op::load) ||
      __ESIMD_DNS::is_device_accessor_with_v<
             AccessorTy, __ESIMD_DNS::accessor_mode_cap::can_write>),
```